Zend/zend_execute.c

```diff
@@ -4354,6 +4355,21 @@ static zend_always_inline uint32_t zend_get_arg_offset_by_name(
 				*(uintptr_t *)(cache_slot + 1) = i;
 				return i;
 			}
+
+			if (fbc->common.attributes != NULL) {
+				zend_attribute *attribute = zend_get_parameter_attribute_str(
+					fbc->common.attributes,
+					"namedparameteralias",
+					sizeof("namedparameteralias")-1,
+					i
+				);
+
+				if (attribute && zend_string_equals(arg_name, Z_STR(attribute->args[0].value) )) {
```

Comment:
A complete implementation of this would support
1. Supporting or forbidding multiple NamedParameterAlias on a single parameter
2. Error checking - (1) check if there is at least one argument, (2) check if the type is a string, (3) possibly attempt to evaluate the argument if the type is IS_CONSTANT_AST and check for `EG(exception)` and return `-1`
3. It may be useful to add some sort of bit flag to the function when compiling to indicate that there are parameter aliases to avoid the need to loop in cases `EXPECTED(*cache_slot == fbc)` doesn't always return early for, such as dynamic calls or polymorphism.

Also, it may be more efficient for the common case to loop over all arguments first and check arg_info->name, and only try to load the attributes in a separate loop if the first attempt failed (efficient for non-variadic functions, but not for variadic functions)